agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/diagnostics/status/StatusFile.java

```diff
@@ -148,39 +131,16 @@ public static String getLogDir() {
 
   private StatusFile() {}
 
-  private static boolean shouldWrite() {
-    if (!DiagnosticsHelper.useAppSvcRpIntegrationLogging()) {
-      return false;
-    }
-    if (writable) {
-      return true;
-    }
-
-    // read-only app services, want to log warning once in this case
-    if (startupLogger != null && !alreadyLogged.getAndSet(true)) {
-      startupLogger.info(
-          "Detected running on a read-only file system. Status json file won't be created. If this is unexpected, please check that process has write access to the directory: {}",
-          directory);
-    }
-    return false;
-  }
-
   public static <T> void putValueAndWrite(String key, T value) {
     putValueAndWrite(key, value, true);
   }
 
   public static <T> void putValueAndWrite(String key, T value, boolean loggingInitialized) {
-    if (!shouldWrite()) {
-      return;
-    }
```

Comment:
why don't we need the `shouldWrite` conditions anymore?

Reply:
canWrite is false negative on linux and doesn't matter where it is. That's why i removed it.

Reply:
added it back.